src/client/java/minicraft/core/Renderer.java

```diff
@@ -185,6 +194,131 @@ public static void render() {
+
+		private AppStatusBar() {}
+
+		private int duration = 120; // Shows for 2 seconds initially.
+
+		private void render() {
+			if (duration == 0) return;
+			MinicraftImage sheet = spriteLinker.getSheet(SpriteType.Gui, "app_status_bar"); // Obtains sheet.
+
+			// Background
+			for (int x = 0; x < 12; ++x) {
+				for (int y = 0; y < 2; ++y) {
+					screen.render(Screen.w - 16 * 8 + x * 8, y * 8, x, y, 0, sheet);
+				}
+			}
+
+			// Hardware Acceleration Status (width = 16)
+			HARDWARE_ACCELERATION_STATUS.render(Screen.w - 16 * 8 + 5, 2, sheet);
+		}
+
+		void tick() {
+			if (duration > 0)
+				duration--;
+			HARDWARE_ACCELERATION_STATUS.tick();
+		}
+
+		void show(int duration) {
+			this.duration = Math.max(this.duration, duration);
+		}
+
+		private void onStatusUpdate() {
+			show(DURATION_ON_UPDATE);
+		}
+
+		private void initialize() {
+			HARDWARE_ACCELERATION_STATUS.initialize();
+		}
+
+		public abstract class AppStatusElement {
+			// width == 16 - size * 2
+			protected final int size; // 0: largest, 1: smaller, etc. (gradually)
+
+			private AppStatusElement(int size) {
+				this.size = size;
+			}
+
+			private static final int BLINK_PERIOD = 10; // 6 Hz
+
+			private int durationUpdated = 0;
+			private boolean blinking = false;
+			private int blinkTick = 0;
+
+			protected void render(int x, int y, MinicraftImage sheet) {
+				if (durationUpdated > 0) {
+					if (blinking) {
+						screen.render(x, y, 10, 3 + size, 0, sheet);
+					}
+				}
+			}
+
+			protected void tick() {
+				if (durationUpdated > 0) {
+					durationUpdated--;
+					if (blinkTick == 0) {
+						blinkTick = BLINK_PERIOD;
+						blinking = !blinking;
+					} else blinkTick--;
+				}
+			}
+
+			protected void updateStatus() {
+				durationUpdated = DURATION_ON_UPDATE;
+				blinking = false;
+				blinkTick = 0;
+				onStatusUpdate();
+			}
+
+			public abstract void updateStatus(int status);
+
+			protected void initialize() {}
+		}
+
+		public class HardwareAccelerationElementStatus extends AppStatusElement {
+			public final int ACCELERATION_ON = 0;
+			public final int ACCELERATION_OFF = 1;
+
+			@MagicConstant(intValues = {ACCELERATION_ON, ACCELERATION_OFF})
+			private int status = ACCELERATION_ON;
+
+			private HardwareAccelerationElementStatus() {
+				super(0);
+			}
+
+			@Override
+			protected void initialize() {
+				status = Boolean.parseBoolean(System.getProperty("sun.java2d.opengl")) ?
+					ACCELERATION_ON : ACCELERATION_OFF;
+			}
+
+			@Override
+			protected void render(int x, int y, MinicraftImage sheet) {
+				super.render(x, y, sheet);
+				if (status == ACCELERATION_ON) {
+					for (int xx = 0; xx < 2; ++xx)
+						screen.render(x + xx * 8, y, xx, 4, 0, sheet);
+				} else {
+					for (int xx = 0; xx < 2; ++xx)
+						screen.render(x + xx * 8, y, 2 + xx, 4, 0, sheet);
+				}
+			}
+
+			@Override
+			public void updateStatus(int status) {
+				super.updateStatus();
+				this.status = status;
+			}
+		}
+	}
```

Comment:
I think `AppStatusBar`, `AppStatusElement` and `HardwareAccelerationElementStatus` are similar enough to to be merged as a single class (similar to how the holding item bar at the bottom of the screen works).

Looking at how `HARDWARE_ACCELERATION_STATUS` is declared on the parent of the object it references, it would make more sense to create a struct with the data you want each icon to have, and then create instances of that struct when adding a new icon.

Blinking and all of that can be implemented in `AppStatusBar`, negating the need to call any other class. And `System.getProperty("sun.java2d.opengl")` doesn't necessarily need to be cached.

Reply:
I'm proposing to move all functionality of this into the `AppStatusBar` class. Bringing in other pull requests into this is irrelevant.

Anything can be implemented in `AppStatusBar`, it's just a question if you are willing to cooperate and do such. Look at how other similar features are implemented (that you haven't designed).